assets/js/popover/view.js

```diff
@@ -93,3 +87,44 @@ module.exports.showResults = function(search, timeSlices, totalCount) {
   analyticsHelper.bindClickDocumentList();
   analyticsHelper.bindClickApp();
 };
+
+var check = null;
+module.exports.search = function search(context, cb) {
+  if(!cb) {
+    cb = function() {};
+  }
+  check = new Date().getTime();
+  var spinner = module.exports.spinner;
+  spinner.start();
+
+  chrome.runtime.sendMessage({
+    type: 'anyfetch::backgroundGetResults',
+    context: context,
+    check: check
+  }, function(response) {
+    spinner.stop();
+    if(response.check === check) {
```

Comment:
This may not be necessary to pass it through the messaging API, but this is to ensure that the results we are displaying are always the latests that the user asked.